### ISSUE TITLE
NullPointerException using ++

### ISSUE THREAD

**Issue by ppgllrd** [nicolasstucki/scala-rrb-vector#2]
I'm using scala-rrb-vector library for benchmarking Scala's Vector vs RRBVectors, but when I run my code I get a NullPointerException:

     Exception in thread "main" java.lang.NullPointerException
    	at scala.collection.immutable.rrbvector.RRBVector.computeBranching(ParRRBVector.scala:901)
    	at scala.collection.immutable.rrbvector.RRBVector.rebalanced(ParRRBVector.scala:748)
    	at scala.collection.immutable.rrbvector.RRBVector.concatenate(ParRRBVector.scala:629)
    	at scala.collection.immutable.rrbvector.RRBVector.$plus$plus(ParRRBVector.scala:508)
    	at Mutate$.apply2(GABenchmark.scala:74)
    	at bitflip$$anonfun$test$1$$anonfun$apply$mcVI$sp$1.apply$mcVI$sp(GABenchmark.scala:95)
    	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:166)
    	at bitflip$$anonfun$test$1.apply$mcVI$sp(GABenchmark.scala:94)
    	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:166)
    	at bitflip$.test(GABenchmark.scala:91)
    	at bitflip$.delayedEndpoint$bitflip$1(GABenchmark.scala:108)
    	at bitflip$delayedInit$body.apply(GABenchmark.scala:78)
    	at scala.Function0$class.apply$mcV$sp(Function0.scala:40)
    	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
    	at scala.App$$anonfun$main$1.apply(App.scala:76)
    	at scala.App$$anonfun$main$1.apply(App.scala:76)
    	at scala.collection.immutable.List.foreach(List.scala:381)
    	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
    	at scala.App$class.main(App.scala:76)
    	at bitflip$.main(GABenchmark.scala:78)
    	at bitflip.main(GABenchmark.scala)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:497)
    	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)

The code leading to this error is the following one:

```scala
import scala.util.Random

object Mutate {
  def apply(rnd : Random, xs : IndexedSeq[Boolean] ): IndexedSeq[Boolean] = {
    val point = rnd.nextInt(xs.length)
    xs.updated(point, !xs(point))
  }

  def apply2(rnd : Random, xs : IndexedSeq[Boolean] ): IndexedSeq[Boolean] = {
    val point = rnd.nextInt(xs.length)
    val (ys,zs) = xs.splitAt(point)
    ys ++ (!zs.head +: zs.tail)
  }
}

object bitflip extends App{
  val iterations = 100000
  val top_length = 4096
  val repeats = 1


  def test(descr : String, init : (Random,Int) => IndexedSeq[Boolean]) {
    var len = 16
    val rnd = new Random(0)
    do {
      val indi = init(rnd,len)

      var times = 0.0
      for(i <- 0 until repeats) {
        val initialTime = System.nanoTime()
        for (i <- 1 until iterations) {
          val newIndi = Mutate.apply2(rnd,indi)
        }
        val finalTime = System.nanoTime()
        times += finalTime - initialTime
      }
      val avg = times/repeats
      println( descr+", "+len+", "+avg/1e9 )
      len = len * 2
    } while ( len <= top_length )
  }

  test("Scala-Vector[Boolean]", (rnd,len) => Vector.fill(len)(rnd.nextDouble > 0.5))

  test("Scala-RRBVector[Boolean]", (rnd,len) => RRBVector.fill(len)(rnd.nextDouble > 0.5))
}
```

The error is raised due to the ++ evaluation in apply2 method. On the other hand, the code executes successfully when tested with Scala standard vectors.

**Comment by nicolasstucki** [comment]
#ppgllrd, this fix has been published on version `0.1.1`. Just update your library dependency to 
```scala
libraryDependencies ++= Seq("io.github.nicolasstucki" %% "scala-rrb-vector" % "0.1.1")
```

**Comment by ppgllrd** [comment]
Thanks
